### PR TITLE
Event tracking

### PR DESCRIPTION
**Pull request by a-szotyori**
Event tracking as described in our [design document](https://docs.google.com/document/d/1iQQD2_XT3U6WuZoFIkd9-J5wyAFxW-mCxWRj6yMYxW8/edit#heading=h.t6baawbhdx7a).

### How to test
Run the screen api example app and look for info logs in logcat for each event. See the `GVLEventTracker` in the screen api example app's `MainActivity` for the log messages.

### Ticket 
[PIA-485](https://ginis.atlassian.net/browse/PIA-485)

### CODE REVIEW

File: ginivision/src/doc/source/event-tracking.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            (...)
+        }
+
+    }
+
+.. note::
+
+     If you use the Screen API all events will be triggered automatically.
+
+     If you use the Component API some events will not be triggered (for ex. events which rely on ``Activity#onBackPressed()``). You need to check whether all the events you are interested in are triggered.
```
Comment (maciejtrybilo) [inline ginivision/src/doc/source/event-tracking.rst]: I might soften the "You need to check" bit and point the reader to where they can check.

File: ginivision/src/test/java/net/gini/android/vision/review/ReviewActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@RunWith(AndroidJUnit4::class)
+class ReviewActivityTest {
+
+    @After
+    fun after() {
+        GiniVision.cleanup(getInstrumentation().targetContext)
+    }
+
+    @Test
+    fun `triggers Exit event when back was pressed`() {
```
Comment (maciejtrybilo) [inline ginivision/src/test/java/net/gini/android/vision/review/ReviewActivityTest.kt]: Back event

File: ginivision/src/test/java/net/gini/android/vision/review/multipage/MultipageReviewActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@LooperMode(LooperMode.Mode.PAUSED)
+class MultipageReviewActivityTest {
+
+    @After
+    fun after() {
+        GiniVision.cleanup(getInstrumentation().targetContext)
+    }
+
+    @Test
+    fun `triggers Exit event when back was pressed`() {
```
Comment (maciejtrybilo) [inline ginivision/src/test/java/net/gini/android/vision/review/multipage/MultipageReviewActivityTest.kt]: Back event

File: ginivision/src/doc/source/event-tracking.rst (hunk 1 above)
Comment (a-szotyori) [reply ginivision/src/doc/source/event-tracking.rst]: Done. Also on second thought I found the table of events you created really useful and added one as well. Had to tweak it to fit and also added a column for the API to quickly see in which API each is triggered automatically.

File: ginivision/src/test/java/net/gini/android/vision/review/ReviewActivityTest.kt (hunk 2 above)
Comment (a-szotyori) [reply ginivision/src/test/java/net/gini/android/vision/review/ReviewActivityTest.kt]: Fixed.

File: ginivision/src/test/java/net/gini/android/vision/review/multipage/MultipageReviewActivityTest.kt (hunk 3 above)
Comment (a-szotyori) [reply ginivision/src/test/java/net/gini/android/vision/review/multipage/MultipageReviewActivityTest.kt]: Fixed.

**Pull request merged**